spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreReactiveOperations.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.firestore;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * An interface of operations that can be done with Cloud Firestore.
+ *
+ * @author Dmitry Solomakha
+ * @since 1.2
+ */
+public interface FirestoreReactiveOperations {
+	/**
+	 * Saves an instance of an object to Cloud Firestore.
+	 * Behaves as update or insert.
+	 * @param <T> the type of the object to save.
+	 * @param instance the instance to save.
+	 * @return the instance that was saved.
+	 */
+	<T> Mono<T> save(T instance);
+
+	/**
+	 * Get all the entities of the given domain type.
+	 * @param <T> the type param of the domain type.
+	 * @param entityClass the domain type to get.
+	 * @return the entities that were found.
+	 */
+	<T> Flux<T> findAll(Class<T> entityClass);
+
+	/**
+	 * Delete all entities of a given domain type.
+	 * @param <T> the type param of the domain type.
+	 * @param entityClass the domain type to delete from Cloud Datastore.
+	 * @return the number of entities that were deleted.
+	 */
+	<T> Mono<Long> deleteAll(Class<T> entityClass);
```

Comment:
Looks like we need a `deleteById` and `findById` to support `ReactiveCrudRepositry`. 

There is a `saveAll`,but i think that could be implemented via `save`, (but would be worth exposing a batch save , i remember they mentioned there's a batch write? )

I think it would be good to add those in this Operations interface so work on the Repo and other sections can begin even before the implementations here are totally done.

Reply:
@dmitry-s WDYT

Reply:
yes definitely, but let's do it in a separate PR